src/merkle_tree.rs

```diff
@@ -0,0 +1,221 @@
+use hmac_sha256::Hash;
+
+pub struct MerkleTree {
+    pub root: Option<[u8; 32]>,
+    pub leafs: Vec<[u8; 32]>,
+}
+
+impl Default for MerkleTree {
+    fn default() -> Self {
+        Self {
+            root: None,
+            leafs: Vec::new(),
+        }
+    }
+}
+
+// TODO: type Hash for [u8; 32]
+// TODO: type Level for Vec<[u8; 32]> of even length (duplicates last element if odd)
```

Comment:
What would be that for?

Reply:
I like having clear types with descriptive names, even if they just "wrap" a simple type. I feel it makes the code clearer to read.

I didn't mean to include the TODO's in the commit tho, they were just ideas for making it better in the future.